distributed/preloading.py

```diff
@@ -77,36 +78,36 @@ def _import_modules(names, file_dir=None):
     Nest dict of names to extracted module interface components if present
     in imported module.
     """
-    result_modules = {}
-
-    for name in names:
-        # import
-        if name.endswith(".py"):
-            # name is a file path
-            if file_dir is not None:
-                basename = os.path.basename(name)
-                copy_dst = os.path.join(file_dir, basename)
-                if os.path.exists(copy_dst):
-                    if not filecmp.cmp(name, copy_dst):
-                        logger.error("File name collision: %s", basename)
-                shutil.copy(name, copy_dst)
-                module = import_file(copy_dst)[0]
-            else:
-                module = import_file(name)[0]
-
+    if "\n" in name:  # not a name, actually the text of the script
```

Comment:
What if it's a one-liner script? Perhaps a safer check would be:

- If it ends in `.py` it's a file to import
- Elif it has no spaces, it's a module name
- Else it's a script to run

Reply:
Nice idea.  Done.